IndexManip/AbstractIndexManip.py

```diff
@@ -5,65 +5,77 @@
 	vincent.jeanselme@gmail.com
 """
 
-class abstractIndexManip:
+import os
+
+class abstractIndexReader:
 	"""
-	Manipulates the index
+	Reads the index
 	Goal is to never have all the indexes in memory, it is why pointers are used
-	for reading
 	Abstract class
 	"""
 
-	def __init__(self):
+	def __init__(self, fileName):
 		"""
-		Creates all the different variables which are used for reading and witting
+		Creates all the different variables which are used for reading
+		Opens the file and initializes pointer in order to read it
 		"""
-		# Signature -> Distinctive signature in order to check the format
-		self.signature = ".index"
-
 		## Reading
 		# Pointer on the file
-		self.ptr = None
+		self.ptr = open(fileName, 'r')
 		# Current read value
 		self.currentValue = None
-
-		## Writing
 		# Name of the file
-		self.source = None
-		# Result of the index computation (create)
-		self.index = None
-		# Save Path
-		self.savePath = None
+		# TODO : Conserve this kind of information in a nother .info file
+		self.source = os.path.basename(fileName)
 
-	def create(self, docTypeFile):
+		# To have a non empty currentValue
+		self.readEntry()
+
+	def readEntry(self):
 		"""
-		Creates the index after extracting the data of the given file
-		Warning docTypeFile has to be adaptated to the current information
-		extraction
+		Reads and returns the next entry in the index (key, value)
 		Abstract
 		"""
 		pass
 
-	def readStart(self, fileName):
+
+class abstractIndexWriter:
+	"""
+	Writes the index
+	Goal is to never have all the indexes in memory, it is why the index is destroyed
+	at the end
+	Abstract class
+	"""
+
+	def __init__(self,docTypeFile):
 		"""
-		Opens the file and initializes pointer in order to read it
-		And reads the first entrance
+		Creates all the different variables which are used for reading and witting
```

Comment:
Little mistake in the comment I think. We are in the **writer** here
Typo on 'witting'